### PR TITLE
Add Autoscaling

### PR DESCRIPTION
**Pull request by NaxAlpha**
This PR allows enabling/disabling and customizing autoscaling for model endpoints in sagemaker inference.

### CODE REVIEW

**Comment by yubozhao** [comment]
It seems like it didn't pass the e2e test with the error:
```File "/home/runner/work/aws-sagemaker-deploy/aws-sagemaker-deploy/./utils/__init__.py", line 49, in run_shell_command
30
    raise Exception(
31
Exception: Failed to run command aws --region us-west-1 cloudformation deploy --stack-name sagemaker-bento-deploy-test-endpoint --template-file /home/runner/work/aws-sagemaker-deploy/aws-sagemaker-deploy/TestService_20211119212312_12639B_sagemaker_deployable/api_template.json --capabilities CAPABILITY_IAM: 
32
Failed to create/update the stack. Run the following command
33
to fetch the list of events leading up to the failure
34
aws cloudformation describe-stack-events --stack-name sagemaker-bento-deploy-test-endpoint```

@NaxAlpha Can you look into it?

cc @jjmachan

File: README.md [hunk 1]
```diff
@@ -82,6 +82,35 @@ You can try out the deployment script with the IrisClassifier for the iris datas
...
+    "skip_stack_deployment": false,
+    "instance_type": "ml.t2.medium",
+    "initial_instance_count": 1,
+    "workers": 3,
+    "timeout": 60,
+    "enable_data_capture": false,
+    "data_capture_s3_prefix": "s3://bucket-name/optional/predix",
+    "data_capture_sample_percent": 100,
+    "scaling": {
+        "disblae": false,
```
Comment (yubozhao) [inline README.md:102]: typo

**Comment by NaxAlpha** [comment]
Hi, @yubozhao Thanks a lot for the review. I am not sure what caused the test to fail, but in my case, what happened was that sometimes it would fail because of some unknown reason, and if I tried again, it would work. I will try to investigate the issue.

**Comment by aarnphm** [comment]
what is the status of this @NaxAlpha ?

**Review by aarnphm — COMMENTED** [review]
everything looks good. Maybe we need to add more tests for this usecase.

**Pull request closed without merging**